src/server/mod.rs

```diff
@@ -60,19 +60,6 @@ pub struct Ack;
 #[derive(Debug)]
 pub struct NoResponse;
 
-/// Empty extra parameters to some request or notification.
-#[derive(Debug, Serialize, PartialEq)]
-pub struct NoParams;
-
-impl<'de> Deserialize<'de> for NoParams {
-    fn deserialize<D>(_deserializer: D) -> Result<NoParams, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        Ok(NoParams)
-    }
-}
```

Comment:
The point in having `NoParams` is for the serialisation and deserialisation, I think that if you use `()` there are bugs both ways (serialised as `()` rather than `{}` or maybe ``, iirc). It's possible some other change means this is no longer an issue though.

Reply:
I will be completely honest here - I'm guilty of relying solely on test suite for `NoParams` change. Now that I look at it they're not covering much messages with `NoParams`, except for `test_shutdown`, but that hands off the already structured `Request`.
I may have jumped gun on that one. I'll discard the last commit for now, thanks for double-checking this!